docs/path-resolution.rst

```diff
@@ -0,0 +1,465 @@
+
+         import "./util.sol" as util; // source unit name: b/util.sol
+
+   - You cannot remap base path or any other part of the path that is only added internally by an
+     import callback:
+
+     .. code-block:: bash
+
+         solc /project/=/contracts/ /project/contract.sol --base-path /project # source unit name: /project/contract.sol
+
+     .. code-block:: solidity
+         :caption: /project/contract.sol
+
+         import "util.sol" as util; // source unit name: util.sol
+
+#. **Target is inserted directly into the source unit name and does not necessarily have to be a valid path.**
+
+   - It can be anything as long as the import callback can handle it.
+     In case of the Host Filesystem Loader this includes also relative paths.
+     When using the JavaScript interface you can even use URLs and abstract identifiers if
+     your callback can handle them.
+
+   - Remapping happens after relative imports have already been resolved into source unit names.
+     This means that targets starting with ``./`` and ``../`` have no special meaning and are
+     relative to the base path rather than to the location of the source file.
+
+   - Remapping targets are not normalized so ``@root/=./a/b//`` will remap ``@root/contract.sol``
+     to ``./a/b//contract.sol`` and not ``a/b/contract.sol``.
+
+   - If the target does not end with a slash, the compiler will not add one automatically:
+
+     .. code-block:: bash
+
+         solc /project/=/contracts /project/contract.sol # source unit name: /project/contract.sol
+
+     .. code-block:: solidity
+         :caption: /project/contract.sol
+
+         import "/project/util.sol" as util; // source unit name: /contractsutil.sol
+
+#. **Context and prefix are patterns and matches must be exact.**
+
+   - ``a//b=c`` will not match ``a/b``.
+   - source unit names are not normalized so ``a/b=c`` will not match ``a//b`` either.
+   - Parts of file and directory names can match as well.
+     ``/newProject/con:/new=old`` will match ``/newProject/contract.sol`` and remap it to
+     ``oldProject/contract.sol``.
+
+#. **At most one remapping is applied to a single import.**
+
+   - If multiple remappings match the same source unit name, the one with the longest matching
+     prefix is chosen.
+   - If prefixes are identical, the one specified last wins.
+   - Remappings do not work on other remappings. For example ``a=b b=c c=d`` will not result in ``a``
+     being remapped to ``d``.
+
+#. **Prefix cannot be empty but context and target are optional.**
+
+   If ``target`` is omitted, it defaults to the value of the ``prefix``.
+
+.. note::
+
+    ``solc`` only allows you to include files from certain directories.
+    They have to be in the directory (or subdirectory) of one of the explicitly specified source
+    files or in the directory (or subdirectory) of a remapping target.
+    If you want to allow direct absolute includes, add the remapping ``/=/``.
```

Comment:
I would say we should state here, that it is not recommended to do.

Reply:
I decided to remove this part after all. It was a part I just copied from the original docs. It's actually pretty weird because you can get the same effect with `--allow-paths /` and using a remapping looks like a roundabout way to do it. Maybe it predates `--allow-paths`?

In any case, we don't really want anyone to do this :)